crates/rattler_package_streaming/src/write.rs

```diff
@@ -0,0 +1,281 @@
+//! Functionality for writing conda packages
+
+use itertools::Itertools;
+
+use std::io::Seek;
+use std::io::Write;
+
+use std::path::Path;
+use std::path::PathBuf;
+
+use bzip2;
+use rattler_conda_types::package::PackageMetadata;
+use tar;
+use zip;
+
+/// a function that sorts paths into two vectors, one that starts with `info/` and one that does not
+/// both vectors are sorted alphabetically for reproducibility
+fn sort_paths(paths: &[PathBuf], base_path: &Path) -> (Vec<PathBuf>, Vec<PathBuf>) {
```

Comment:
You could also take `impl IntoIterator<Item=impl AsRef<Path>>` as the input type and return two iterators instead of directly converting to vecs.